tensorboard/webapp/runs/store/runs_selectors_utils.ts

```diff
@@ -0,0 +1,137 @@
+  const metricDisplayNames = new Map<string, Set<string>>();
+  const validationError: string[] = [];
+
+  for (const spec of specs) {
+    for (const hparamSpec of spec.hparams) {
+      if (!hparamDisplayNames.has(hparamSpec.name)) {
+        hparamDisplayNames.set(hparamSpec.name, new Set());
+      }
+      hparamDisplayNames.get(hparamSpec.name)!.add(hparamSpec.displayName);
+      if (!hparams.has(hparamSpec.name)) {
+        hparams.set(hparamSpec.name, {...hparamSpec});
+      } else {
+        const specA = hparams.get(hparamSpec.name)!;
+        const specB = hparamSpec;
+        if (specA.type !== specB.type) {
+          validationError.push(
+            `Hparam, ${specB.name}, types have to match. Got: ` +
+              `${specA.type} vs. ${specB.type}`
+          );
+        }
+
+        if (
+          specA.domain.type === DomainType.INTERVAL &&
+          specB.domain.type === DomainType.INTERVAL
+        ) {
+          if (
+            specA.domain.minValue !== specB.domain.minValue ||
+            specA.domain.maxValue !== specB.domain.maxValue
+          ) {
+            validationError.push(
+              `Hparam, ${specB.name}, domains have to match. ` +
+                `Got: ${specA.domain} vs. ${specB.domain}`
+            );
+          }
+        } else if (
+          specA.domain.type === DomainType.DISCRETE &&
+          specB.domain.type === DomainType.DISCRETE
+        ) {
+          const valueSet = new Set([
+            ...specA.domain.values,
+            ...specB.domain.values,
+          ]);
+          if (
+            specA.domain.values.length !== specB.domain.values.length ||
+            specA.domain.values.length !== valueSet.size
+          ) {
+            validationError.push(
+              `Hparam, ${specB.name}, domains have to match. ` +
+                `Got: ${specA.domain} vs. ${specB.domain}`
+            );
+          }
+        } else {
+          validationError.push(
+            `Hparam, ${specB.name}, domains have to match. ` +
+              `Got: ${specA.domain} vs. ${specB.domain}`
+          );
+        }
+      }
+    }
+
+    for (const metricSpec of spec.metrics) {
+      if (!metricDisplayNames.has(metricSpec.tag)) {
+        metricDisplayNames.set(metricSpec.tag, new Set());
+      }
+      metricDisplayNames.get(metricSpec.tag)!.add(metricSpec.displayName);
+      if (!metrics.has(metricSpec.tag)) {
+        metrics.set(metricSpec.tag, {...metricSpec});
+      } else {
+        const specA = metrics.get(metricSpec.tag)!;
+        const specB = metricSpec;
+        if (specA.datasetType !== specB.datasetType) {
+          validationError.push(
+            `Metric, ${specB.tag}, datasetTypes have to match. Got: ` +
+              `${specA.datasetType} vs. ${specB.datasetType}`
+          );
+        }
+      }
+    }
+  }
+
+  if (validationError.length) {
+    // TODO(b/157733179): handle the error gracefully on the view side.
```

Comment:
I assume this is an intended change?

Reply:
Yes, this line is intended.  It would be good to perhaps move all internal bug #s into new GitHub issues, but that can be done independently outside of this PR series.

runs_selectors_utils.ts was not manually altered in this PR's manual changes:
https://github.com/tensorflow/tensorboard/pull/4181/files/9ae8c76d4e128d1054b2cc138e250eba42b27a97..da8db23b5648e8d5268c1401f5c4b97d9e2a3bfc